### ISSUE TITLE
[SF][0.25.5][useSyncRef] The callback ref used to measure a size of a DOM node will be called unnecessarily

### ISSUE THREAD

**Issue by i323808** [SAP/ui5-webcomponents-react#3316]
**Describe the bug**
The callback ref implemented by useSyncRef will be called again and again when re-render the component. And the state manipulated by the callback ref is set incorrect value during re-rendering. The callback ref looks like below:
const measuredRef = useCallback((node) => {
    if (node !== null) {
      setIsStartColumnVisible(node.startColumnVisible);
    }
  }, []);
By the way, the older version hook named useConsolidatedRef has no this issue.

**Isolated Example**
*Please provide an isolated example if possible by forking [this codesandbox](https://codesandbox.io/s/ui5-webcomponents-react-template-29l34).*

**To Reproduce**

**Expected behavior**
The callback ref used to measure the size of DOM node is supposed to be called only when the component mounts and unmounts.

**Screenshots**
![useSyncRef](https://user-images.githubusercontent.com/30744017/182078563-a92a1d09-f213-4b68-a99a-df4529487993.png)

**UI5 Web Components for React Information**
`@ui5/webcomponents` version: 1.4.0
`@ui5/webcomponents-react` version: 0.25.5
Operating System: macOS Monterey
Browser: Chrome Version 103.0.5060.134 (Official Build) (x86_64)

**Additional context**

**Comment by MarcusNotheis** [comment]
Thanks for reporting!

Actually, I think the behaviour of calling the callback ref on every render of the component is correct, you can compare it with this "plain" react sandbox: https://codesandbox.io/s/black-dawn-lq9y9k?file=/src/App.js

Regarding the issue of incorrect values: I would assume that the callback ref is called while the `FlexibleColumnLayout` is not updated yet. Can you try to wrap the content of your `useCallback` with `setTimeout(() => {}, 0)` or with `requestAnimationFrame()`?

**Comment by i323808** [comment]
Hi @MarcusNotheis ,

Thanks for your suggestion. The `requestAnimationFrame` workaround is working fine.

By the way, is it possible to wrap the measured Ref with `useCallback` as React recommended(https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node)? Now, `useSyncRef` does not use `useCallback`, though app is using `useCallback` likes below:
```
const measuredRef = useCallback((node) => {
	if (node !== null) {
		setIsStartColumnVisible(node.startColumnVisible);
	}
}, []);
```
Actually, this is the root cause of this issue. Previously, `useConsolidatedRef` has dependency(https://github.com/SAP/ui5-webcomponents-react/blob/rel/0.131/packages/base/src/hooks/useConsolidatedRef.ts#L17). But `useSyncRef` has no any dependency, so the callback will be called once the component is repainted.

Thanks,
Jane

**Comment by MarcusNotheis** [comment]
Sorry for the late reply, I somehow missed your response.
You're right, we're missing a `useCallback` here, will be fixed with #3381.
